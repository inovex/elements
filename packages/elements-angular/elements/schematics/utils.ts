import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
