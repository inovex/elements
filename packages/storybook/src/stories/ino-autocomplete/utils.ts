import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
