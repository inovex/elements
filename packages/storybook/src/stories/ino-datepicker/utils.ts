import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
