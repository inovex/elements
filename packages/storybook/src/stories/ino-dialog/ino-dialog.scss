ino-button {
  .button__label {
    pointer-events: none;
  }
}

#demo-custom-content-dialog {
  .customizable-carousel {
    display: block;
    width: 100%;
    height: 20rem;
  }

  header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 2rem;
  }

  header ino-icon {
    --icon-width: 2rem;
    --icon-height: 2rem;
  }

  header h3 {
    margin: 0;
  }

  footer {
    margin-top: 2rem;
    display: flex;
    justify-content: space-between;
  }
}
