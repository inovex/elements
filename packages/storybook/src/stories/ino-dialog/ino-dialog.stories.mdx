import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-dialog/readme.md";

<Meta title="Structure/ino-dialog/Documentation" />

<Description markdown={readme} />
