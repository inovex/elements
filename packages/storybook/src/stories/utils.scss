/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
