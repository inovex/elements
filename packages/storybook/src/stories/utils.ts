import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
