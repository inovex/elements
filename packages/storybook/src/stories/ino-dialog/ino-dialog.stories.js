import { storiesOf } from '@storybook/html';

import componentReadme from '_local-elements/src/components/ino-dialog/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import addons from '@storybook/addons';
import CoreEvents from '@storybook/core-events';
import './ino-dialog.scss';

function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function(e) {
    const el = e.target;
    console.log(el);
    if (el.id === 'open-dialog-btn') {
      const dialog = document.getElementById('demo-dialog-1');
      dialog.inoOpen = true;
    }
    if(el.id === 'close-dialog-btn') {
      const dialog = document.getElementById('demo-dialog-1');
      dialog.inoOpen = false;
    }
  };

  const checkedChangeHandler = function(e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-segment-button') {
      return;
    }
    const group = el.closest('ino-segment-group');
    group.setAttribute('value', el.getAttribute('value'));
  };

  document.addEventListener('checkedChange', checkedChangeHandler);

  document.addEventListener('click', eventHandler);
  document.addEventListener('clickEl', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('click', eventHandler);
    document.removeEventListener('clickEl', eventHandler);
    document.removeEventListener('checkedChange', checkedChangeHandler);
  };
}

storiesOf('structure/<ino-dialog>', module)
  .addDecorator(withStencilReadme(componentReadme))
  .addDecorator(story => {
    addons
      .getChannel()
      .emit(CoreEvents.REGISTER_SUBSCRIPTION, subscribeToComponentEvents);
    return story();
  })
  .add('Default usage', () => /*html*/`

     <ino-button id="open-dialog-btn">Open Dialog</ino-button>
     <ino-dialog id="demo-dialog-1">
        <div class="ino-dialog-header" slot="header">
            <div class="ino-dialog-title">
                <h3>Select your favourite characters:</h3>
            </div>
            <ino-icon-button ino-icon="close" id="close-dialog-btn"></ino-icon-button>
        </div>
        <div class="ino-dialog-content">
            <ino-segment-group value="2" id="dense-segment-grp">
              <ino-segment-button ino-dense value="1">Morty</ino-segment-button>
              <ino-segment-button ino-dense value="2">Pickle Rick</ino-segment-button>
              <ino-segment-button ino-dense value="3">Squanchy</ino-segment-button>
              <ino-segment-button ino-dense value="4">Beth</ino-segment-button>
              <ino-segment-button ino-dense value="5">Jerry</ino-segment-button>
              <ino-segment-button ino-dense value="6">Summer</ino-segment-button>
          </ino-segment-group>
        </div>
        <div class="ino-dialog-footer" slot="footer">
          <ino-button ino-icon="remove" id="close-dialog-btn" style="--button-color-primary: red; --button-color-primary-light: rgba(255, 0, 0, 0.7); --button-color-primary-dark: red">Cancel</ino-button>
          <ino-button ino-icon="mail" id="close-dialog-btn">Submit</ino-button>
        </div>
     </ino-dialog>
  `);
