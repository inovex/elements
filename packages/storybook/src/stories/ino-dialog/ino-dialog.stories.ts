import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-dialog.scss';

export default {
  title: 'Structure/<ino-dialog>',
  component: 'ino-dialog',
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const openDialogHandler = function (e) {
          const el = e.target;

          if (el.id === 'open-dialog-btn') {
            const dialog = document.getElementById(
              'demo-dialog-1'
            ) as HTMLInoDialogElement;
            dialog.open = true;
          }
          if (el.id === 'close-dialog-btn') {
            const dialog = document.getElementById(
              'demo-dialog-1'
            ) as HTMLInoDialogElement;
            dialog.open = false;
          }
        };

        const checkedChangeHandler = function (e) {
          if (e.target.tagName.toLowerCase() !== 'ino-segment-button') {
            return;
          }
          const segmentButton: HTMLInoSegmentButtonElement = e.target;
          const group: HTMLInoSegmentGroupElement = segmentButton.closest(
            'ino-segment-group'
          );
          group.value = segmentButton.value;
        };

        document.addEventListener('checkedChange', checkedChangeHandler);

        document.addEventListener('click', openDialogHandler);
        document.addEventListener('clickEl', openDialogHandler);

        return () => {
          document.removeEventListener('click', openDialogHandler);
          document.removeEventListener('clickEl', openDialogHandler);
          document.removeEventListener('checkedChange', checkedChangeHandler);
        };
      });

      return story();
    },
  ],
};

export const Playground: Story<Components.InoDialog> = (args) => html`
  <ino-button id="open-dialog-btn">Open Dialog</ino-button>
  <ino-dialog id="demo-dialog-1" open="${args.open}">
    <div class="ino-dialog-header" slot="header">
      <div class="ino-dialog-title">
        <h5>Select your favourite characters:</h5>
      </div>
      <ino-icon-button icon="close" id="close-dialog-btn"></ino-icon-button>
    </div>
    <div class="ino-dialog-content">
      <ino-segment-group value="2" id="dense-segment-grp">
        <ino-segment-button dense value="1">Morty</ino-segment-button>
        <ino-segment-button dense value="2">Pickle Rick</ino-segment-button>
        <ino-segment-button dense value="3">Squanchy</ino-segment-button>
        <ino-segment-button dense value="4">Beth</ino-segment-button>
        <ino-segment-button dense value="5">Jerry</ino-segment-button>
        <ino-segment-button dense value="6">Summer</ino-segment-button>
      </ino-segment-group>
    </div>
    <div class="ino-dialog-footer" slot="footer">
      <ino-button
        icon="remove"
        id="close-dialog-btn"
        style="--button-color-primary: red; --button-color-primary-light: rgba(255, 0, 0, 0.7); --button-color-primary-dark: red"
        >Cancel
      </ino-button>
      <ino-button icon="mail" id="close-dialog-btn">Submit</ino-button>
    </div>
  </ino-dialog>
`;

Playground.args = {
  open: false,
};
