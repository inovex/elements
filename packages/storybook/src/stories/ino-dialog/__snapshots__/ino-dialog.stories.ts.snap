// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/<ino-dialog> ClosingAction smoke-test 1`] = `
<div id="root-inner">
  <div class="story-dialog">
    <ino-button class="open-dialog-btn hydrated"
                data-dialog-id="demo-action-dialog"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Open Dialog
        </span>
      </button>
    </ino-button>
    <ino-dialog id="demo-action-dialog"
                open="false"
                fullwidth="false"
                dismissible="true"
                attach-to=".story-dialog"
                class="hydrated"
    >
      <div class="ino-dialog-header">
        <div class="ino-dialog-title">
          <h3>
            Select your favourite characters
          </h3>
        </div>
        <ino-icon-button icon="close"
                         data-ino-dialog-action="close"
                         class="ino-icon-button--color-scheme-primary hydrated"
        >
          <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                  type="button"
                  style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
          >
            <div class="mdc-icon-button__ripple">
            </div>
            <span class="mdc-icon-button__icon">
              <ino-icon class="ino-icon--color-scheme-primary hydrated">
              </ino-icon>
            </span>
          </button>
        </ino-icon-button>
      </div>
      <div class="ino-dialog-content">
        <ino-segment-group value="2"
                           id="ino-segment-group-id__kyyfmd0hi"
                           class="hydrated"
        >
          <ino-segment-button dense
                              value="1"
                              class="hydrated"
          >
            Morty
          </ino-segment-button>
          <ino-segment-button dense
                              value="2"
                              class="hydrated ino-segment-button--checked"
                              checked
          >
            Pickle Rick
          </ino-segment-button>
          <ino-segment-button dense
                              value="3"
                              class="hydrated"
          >
            Squanchy
          </ino-segment-button>
          <ino-segment-button dense
                              value="4"
                              class="hydrated"
          >
            Beth
          </ino-segment-button>
          <ino-segment-button dense
                              value="5"
                              class="hydrated"
          >
            Jerry
          </ino-segment-button>
          <ino-segment-button dense
                              value="6"
                              class="hydrated"
          >
            Summer
          </ino-segment-button>
        </ino-segment-group>
      </div>
      <div class="ino-dialog-footer">
        <ino-button icon="remove"
                    data-ino-dialog-action="cancel"
                    fill="outline"
                    color-scheme="secondary"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Cancel
            </span>
          </button>
        </ino-button>
        <ino-button icon="mail"
                    data-ino-dialog-action="submit"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Submit
            </span>
          </button>
        </ino-button>
      </div>
    </ino-dialog>
  </div>
</div>
`;

exports[`Structure/<ino-dialog> Dismissible smoke-test 1`] = `
<div id="root-inner">
  <div class="story-dialog">
    <ino-button class="open-dialog-btn hydrated"
                data-dialog-id="undefined"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Open Dialog
        </span>
      </button>
    </ino-button>
    <ino-dialog id="undefined"
                open="false"
                fullwidth="false"
                dismissible="true"
                attach-to=".story-dialog"
                class="hydrated"
    >
      <div class="ino-dialog-header">
        <div class="ino-dialog-title">
          <h3>
            Select your favourite characters
          </h3>
        </div>
        <ino-icon-button icon="close"
                         data-ino-dialog-action="close"
                         class="ino-icon-button--color-scheme-primary hydrated"
        >
          <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                  type="button"
                  style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
          >
            <div class="mdc-icon-button__ripple">
            </div>
            <span class="mdc-icon-button__icon">
              <ino-icon class="ino-icon--color-scheme-primary hydrated">
              </ino-icon>
            </span>
          </button>
        </ino-icon-button>
      </div>
      <div class="ino-dialog-content">
        <ino-segment-group value="2"
                           id="ino-segment-group-id__4avr616ok"
                           class="hydrated"
        >
          <ino-segment-button dense
                              value="1"
                              class="hydrated"
          >
            Morty
          </ino-segment-button>
          <ino-segment-button dense
                              value="2"
                              class="hydrated ino-segment-button--checked"
                              checked
          >
            Pickle Rick
          </ino-segment-button>
          <ino-segment-button dense
                              value="3"
                              class="hydrated"
          >
            Squanchy
          </ino-segment-button>
          <ino-segment-button dense
                              value="4"
                              class="hydrated"
          >
            Beth
          </ino-segment-button>
          <ino-segment-button dense
                              value="5"
                              class="hydrated"
          >
            Jerry
          </ino-segment-button>
          <ino-segment-button dense
                              value="6"
                              class="hydrated"
          >
            Summer
          </ino-segment-button>
        </ino-segment-group>
      </div>
      <div class="ino-dialog-footer">
        <ino-button icon="remove"
                    data-ino-dialog-action="cancel"
                    fill="outline"
                    color-scheme="secondary"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Cancel
            </span>
          </button>
        </ino-button>
        <ino-button icon="mail"
                    data-ino-dialog-action="submit"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Submit
            </span>
          </button>
        </ino-button>
      </div>
    </ino-dialog>
  </div>
</div>
`;

exports[`Structure/<ino-dialog> FullWidth smoke-test 1`] = `
<div id="root-inner">
  <div class="story-dialog">
    <ino-button class="open-dialog-btn hydrated"
                data-dialog-id="undefined"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Open Dialog
        </span>
      </button>
    </ino-button>
    <ino-dialog id="undefined"
                open="false"
                fullwidth="true"
                dismissible="false"
                attach-to=".story-dialog"
                class="ino-dialog--fullwidth hydrated"
    >
      <div class="ino-dialog-header">
        <div class="ino-dialog-title">
          <h3>
            Select your favourite characters
          </h3>
        </div>
        <ino-icon-button icon="close"
                         data-ino-dialog-action="close"
                         class="ino-icon-button--color-scheme-primary hydrated"
        >
          <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                  type="button"
                  style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
          >
            <div class="mdc-icon-button__ripple">
            </div>
            <span class="mdc-icon-button__icon">
              <ino-icon class="ino-icon--color-scheme-primary hydrated">
              </ino-icon>
            </span>
          </button>
        </ino-icon-button>
      </div>
      <div class="ino-dialog-content">
        <ino-segment-group value="2"
                           id="ino-segment-group-id__i8a1j2mpx"
                           class="hydrated"
        >
          <ino-segment-button dense
                              value="1"
                              class="hydrated"
          >
            Morty
          </ino-segment-button>
          <ino-segment-button dense
                              value="2"
                              class="hydrated ino-segment-button--checked"
                              checked
          >
            Pickle Rick
          </ino-segment-button>
          <ino-segment-button dense
                              value="3"
                              class="hydrated"
          >
            Squanchy
          </ino-segment-button>
          <ino-segment-button dense
                              value="4"
                              class="hydrated"
          >
            Beth
          </ino-segment-button>
          <ino-segment-button dense
                              value="5"
                              class="hydrated"
          >
            Jerry
          </ino-segment-button>
          <ino-segment-button dense
                              value="6"
                              class="hydrated"
          >
            Summer
          </ino-segment-button>
        </ino-segment-group>
      </div>
      <div class="ino-dialog-footer">
        <ino-button icon="remove"
                    data-ino-dialog-action="cancel"
                    fill="outline"
                    color-scheme="secondary"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Cancel
            </span>
          </button>
        </ino-button>
        <ino-button icon="mail"
                    data-ino-dialog-action="submit"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Submit
            </span>
          </button>
        </ino-button>
      </div>
    </ino-dialog>
  </div>
</div>
`;

exports[`Structure/<ino-dialog> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-dialog">
    <ino-button class="open-dialog-btn hydrated"
                data-dialog-id="undefined"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Open Dialog
        </span>
      </button>
    </ino-button>
    <ino-dialog id="undefined"
                open="false"
                fullwidth="false"
                dismissible="false"
                attach-to=".story-dialog"
                class="hydrated"
    >
      <div class="ino-dialog-header">
        <div class="ino-dialog-title">
          <h3>
            Select your favourite characters
          </h3>
        </div>
        <ino-icon-button icon="close"
                         data-ino-dialog-action="close"
                         class="ino-icon-button--color-scheme-primary hydrated"
        >
          <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                  type="button"
                  style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
          >
            <div class="mdc-icon-button__ripple">
            </div>
            <span class="mdc-icon-button__icon">
              <ino-icon class="hydrated ino-icon--color-scheme-primary">
              </ino-icon>
            </span>
          </button>
        </ino-icon-button>
      </div>
      <div class="ino-dialog-content">
        <ino-segment-group value="2"
                           id="ino-segment-group-id__ds06pakmn"
                           class="hydrated"
        >
          <ino-segment-button dense
                              value="1"
                              class="hydrated"
          >
            Morty
          </ino-segment-button>
          <ino-segment-button dense
                              value="2"
                              class="hydrated ino-segment-button--checked"
                              checked
          >
            Pickle Rick
          </ino-segment-button>
          <ino-segment-button dense
                              value="3"
                              class="hydrated"
          >
            Squanchy
          </ino-segment-button>
          <ino-segment-button dense
                              value="4"
                              class="hydrated"
          >
            Beth
          </ino-segment-button>
          <ino-segment-button dense
                              value="5"
                              class="hydrated"
          >
            Jerry
          </ino-segment-button>
          <ino-segment-button dense
                              value="6"
                              class="hydrated"
          >
            Summer
          </ino-segment-button>
        </ino-segment-group>
      </div>
      <div class="ino-dialog-footer">
        <ino-button icon="remove"
                    data-ino-dialog-action="cancel"
                    fill="outline"
                    color-scheme="secondary"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Cancel
            </span>
          </button>
        </ino-button>
        <ino-button icon="mail"
                    data-ino-dialog-action="submit"
                    class="hydrated"
        >
          <button class="button button__variant--filled"
                  type="button"
          >
            <span class="button__label">
              Submit
            </span>
          </button>
        </ino-button>
      </div>
    </ino-dialog>
  </div>
</div>
`;
